cli/io/spec_storage.py

```diff
@@ -159,3 +161,39 @@ def run_spec_storage(
             f" -s {benchmark}-{self.outputlog}-{last_path_component}"
         )
         return self.execute(sudo=True, long_running=True, cmd=cmd)
+
+    def append_to_csv(self, output_file, metrics):
+        fieldnames = list(metrics.keys())
+        file_exists = os.path.isfile(output_file)
+
+        with open(output_file, "a", newline="") as csvfile:
+            writer = csv.DictWriter(csvfile, fieldnames=fieldnames)
+            if not file_exists:
+                writer.writeheader()
+            writer.writerow(metrics)
+
+    def extract_metrics(self, remote_file, **kwargs):
+        # Create an ElementTree object from the remote file
+        tree = ET.parse(remote_file)
+        root = tree.getroot()
+
+        metrics = kwargs
+        business_metric = root.find(".//business_metric").text
+        metrics["business_metric"] = business_metric
+        benchmark = root.find(".//benchmark").attrib["name"]
+        metrics["benchmark"] = benchmark
+        for metric in root.findall(".//metric"):
+            name = metric.attrib["name"]
```

Comment:
Can we also include units with metric name like for,

-  throughput - KB/s or MB/s or Gbps as applicable
-  'secs' for metrics such as latency
- KB/MB/GB for size